### DISCUSSION TITLE
Quesitons about max volumes and ec shards

### DISCUSSION THREAD

**Discussion by valan** [seaweedfs/seaweedfs#6372] [Q&A] [answered]
Hi, I have a few questions about the way things work with max volumes, ec shards, and disk space:

1. is there any way to have automatic volume max (`-max=0`) but leave some extra room? For example, have it choose automatically but set it 10 volumes less than it normally would to leave an extra 300GiB, or even better set the desired free space directly?
2. If max volumes count is exceeded on a volume server, do volumes or ec shards need to be moved to other volume servers manually (or with out of band process)?
3. What is the best way to move ec shards off of a volume server that is getting full?
4. Does ec.balance avoid placing shards on volume servers with exceeded volume counts?
5. Do 10 ec shards on a volume server essentially count as one volume slot on a volume server?

Thanks

**Comment by chrislusf** [comment]
cc @proton-lisandro-pin who re-wrote the ec balance logic recently.

1. need a code change.
2. not necessarily. Writable volumes on the volume servers would be come readonly.
3. current ec balance may not have handled this. cc @proton-lisandro-pin 
4. cc @proton-lisandro-pin 
5. yes

**Comment by ghost** [comment, marked as answer]
> What is the best way to move ec shards off of a volume server that is getting full?

I reckon `ec.balance` should do this automatically for you. It will prioritize racks/servers with lower utilization when deciding where to relocate EC shards.

> Does ec.balance avoid placing shards on volume servers with exceeded volume counts?

Yes.

**Reply by chrislusf** [reply]
EC placement/balance algo needs a wiki page.

**Reply by ghost** [reply]
> EC placement/balance algo needs a wiki page.

+1. It's coming as soon as https://github.com/seaweedfs/seaweedfs/discussions/6179 is wrapped.

**Reply by ghost** [reply]
https://github.com/seaweedfs/seaweedfs/wiki/Erasure-Coding-for-warm-storage#ec-data-balancing is now updated.
